core/src/main/java/org/bitcoinj/wallet/BipStandardKeyChainGroupStructure.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright by the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bitcoinj.wallet;
+
+import org.bitcoinj.crypto.HDPath;
+import org.bitcoinj.params.MainNetParams;
+import org.bitcoinj.script.Script;
+
+/**
+ * KeyChainGroupStructure that supports BIP44, BIP84, etc.
+ */
+public interface BipStandardKeyChainGroupStructure extends KeyChainGroupStructure {
+    /**
+     *  Map desired output script type to an account path.
+     *  Default to MainNet and account 0.
+     * @param outputScriptType the script/address type
+     * @return account path
+     */
+    @Override
+    default HDPath accountPathFor(Script.ScriptType outputScriptType) {
```

Comment:
I still need to wrap my head around the new Java8 features.

This default method seems to be quite generic and not related to BIP-44/49/84 specifically. So wouldn't it make sense to add the default method to `KeyChainGroupStructure` instead and maybe not subclass the interface at all?

Reply:
I guess the subclassing is because otherwise the change would be API breaking?

Reply:
I quickly liked these from the airport last week and commented on the main thread of this PR that, yeah I did it this way to be non-breaking, but think we should consider moving everything into `KeyChainGroupStructure` and making the breaking change.

Reply:
msgilligan:msgilligan-BipStandardKeyChainGroupStructure

Reply:
OK, I force-pushed a new implementation with everything in `KeyChainGroupStructure` (i.e. without a subclass.)